source/fuzz/fuzzer_pass_apply_id_synonyms.cpp

```diff
@@ -85,6 +85,12 @@ void FuzzerPassApplyIdSynonyms::Apply() {
         auto synonym_to_try = synonyms_to_try[synonym_index];
         synonyms_to_try.erase(synonyms_to_try.begin() + synonym_index);
 
+        if (!fuzzerutil::IdIsAvailableAtUse(GetIRContext(), use_inst,
```

Comment:
I suggest you restructure the following as:

if (synonym's index size > 0 && cannot insert instruction before opcode && opcode is not OpPhi) {
  continue;
}

if (id is not available at use) {
  continue;
}

Then start doing what is currently at line 131: "// We either replace the use with an id known to be synonymous (when"

You can then fold your new work on OpPhi into the 'else' branch that currently starts on line 139: it should be possible to just apply the composite extract with a different instruction descriptor depending on whether or not the use instruction is an OpPhi.  In the case that it is an OpPhi you need to get an instruction descriptor for the parent block's terminator (the code for getting the parent block looks good).